src/Google/GoogleHelper.php

```diff
@@ -0,0 +1,1350 @@
+	 *
+	 * WooCommerce Languages -> https://translate.wordpress.org/projects/wp-plugins/woocommerce/
+	 * Google Supported Languages -> https://support.google.com/merchants/answer/160637?hl=en
+	 * ISO 639-1 -> https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
+	 *
+	 * @return array
+	 */
+	public function get_mc_supported_languages(): array {
+		// Repeated values removed:
+		// 'pt', // Brazilian Portuguese
+		// 'zh', // Simplified Chinese*
+
+		return [
+			'ar' => 'ar', // Arabic
+			'cs' => 'cs', // Czech
+			'da' => 'da', // Danish
+			'nl' => 'nl', // Dutch
+			'en' => 'en', // English
+			'fi' => 'fi', // Finnish
+			'fr' => 'fr', // French
+			'de' => 'de', // German
+			'he' => 'he', // Hebrew
+			'hu' => 'hu', // Hungarian
+			'id' => 'id', // Indonesian
+			'it' => 'it', // Italian
+			'ja' => 'ja', // Japanese
+			'ko' => 'ko', // Korean
+			'el' => 'el', // Modern Greek
+			'nb' => 'nb', // Norwegian (Norsk Bokmål)
+			'nn' => 'nn', // Norwegian (Norsk Nynorsk)
+			'no' => 'no', // Norwegian
+			'pl' => 'pl', // Polish
+			'pt' => 'pt', // Portuguese
+			'ro' => 'ro', // Romanian
+			'ru' => 'ru', // Russian
+			'sk' => 'sk', // Slovak
+			'es' => 'es', // Spanish
+			'sv' => 'sv', // Swedish
+			'th' => 'th', // Thai
+			'zh' => 'zh', // Traditional Chinese
+			'tr' => 'tr', // Turkish
+			'uk' => 'uk', // Ukrainian
+			'vi' => 'vi', // Vietnamese
+		];
+	}
+
+	/**
+	 * Get whether the country is supported by the Merchant Center.
+	 *
+	 * @param string $country Country code.
+	 *
+	 * @return bool True if the country is in the list of MC-supported countries.
+	 */
+	public function is_country_supported( string $country ): bool {
+		return array_key_exists(
+			strtoupper( $country ),
+			$this->get_mc_supported_countries_data()
+		);
+	}
+
+	/**
+	 * Maps the ID of the Merchant Center supported locations to their ISO 3166-1/ISO 3166-2 codes.
+	 */
+	protected function map_location_id_to_code(): void {
+		$this->location_id_code_map = [];
+
+		$countries = $this->get_mc_supported_countries_data();
+		foreach ( $countries as $country ) {
+			$this->location_id_code_map[ $country['id'] ] = $country['code'];
+		}
+
+		foreach ( self::COUNTRY_SUBDIVISIONS as $subdivisions ) {
+			foreach ( $subdivisions as $item ) {
+				$this->location_id_code_map[ $item['id'] ] = $item['code'];
+			}
+		}
+	}
+
+	/**
+	 * Find the ISO 3166-1 code of the Merchant Center supported country by its location ID.
+	 *
+	 * @param int $id
+	 *
+	 * @return string|null ISO 3166-1 representation of the country code.
+	 */
+	public function find_country_code_by_id( int $id ): ?string {
+		if ( ! isset( $this->location_id_code_map ) ) {
+			$this->map_location_id_to_code();
+		}
+
+		return $this->location_id_code_map[ $id ] ?? null;
+	}
+
+	/**
+	 * Find the ISO 3166-2 code of the Merchant Center supported subdivision by its location ID.
+	 *
+	 * @param int $id
+	 *
+	 * @return string|null ISO 3166-2 representation of the country code.
```

Comment:
As it is mentioned in the comment #1282 (comment) , this is not really compliant with ISO 3166-2 so it could be convenient to update this comment. Also, we need to adjust: "the representation of country code" with the subdivision code.

Reply:
Removed in https://github.com/woocommerce/google-listings-and-ads/pull/1282/commits/9111f9414eec2b137f68772fd5f0f6e482d9317b